ui/src/view/button.rs

```diff
@@ -28,6 +46,26 @@ where
     T: Clone,
 {
     pub fn view(self) -> Node<T> {
-        button![self.label, ev(Ev::Click, |_| self.on_click)]
+        let tag = match self.on_click {
+            Handler(_) => "button",
+            Route(_) => "a",
+        };
+
+        let mut element: El<T> = El::empty(Tag::Custom(Cow::Borrowed(tag)));
+
+        element.children.push(text(self.label));
+
+        match self.on_click {
+            Handler(msg) => {
+                element.add_event_handler(ev(Ev::Click, |_| msg));
+            }
+            Route(route) => {
+                element.add_attr(Cow::Borrowed("href"), route.as_str());
+            }
```

Comment:
If its a click event kind of button use an event handler with our `msg`, otherwise give it an `href` attribute.